M/MAGMA/build_tarballs.jl

```diff
@@ -1,38 +1,56 @@
+# Note that this script can accept some limited command-line arguments, run
+# `julia build_tarballs.jl --help` to see a usage message.
 using BinaryBuilder, Pkg
 
+include(joinpath(@__DIR__, "..", "..", "platforms", "cuda.jl"))
+
 name = "MAGMA"
-version = v"2.6.2"
+version = v"2.7.0"
+
+# Collection of sources required to complete build
 sources = [
-    GitSource("https://bitbucket.org/icl/magma", "864fa8b28e8c2905a41fbef32f156695378dc104"),
+    ArchiveSource("http://icl.utk.edu/projectsfiles/magma/downloads/magma-2.7.0.tar.gz", "fda1cbc4607e77cacd8feb1c0f633c5826ba200a018f647f1c5436975b39fd18"),
     DirectorySource("./bundled")
 ]
 
 # Bash recipe for building across all platforms
 script = raw"""
-cd $WORKSPACE/srcdir/magma/
-
+cd $WORKSPACE/srcdir/magma-*
+export CUDADIR=${WORKSPACE}/destdir/cuda
+export PATH=${PATH}:${CUDADIR}
 cp ../make.inc .
-
-make -j${nproc} sparse-lib
+# reduce parallelism since otherwise the builder may OOM.
+(( nproc=1+nproc/3 ))
+make -j${nproc} sparse-shared
 make install prefix=${prefix}
 install_license COPYRIGHT
 """
 
-# These are the platforms we will build for by default, unless further
-# platforms are passed in on the command line
-platforms = [Platform("x86_64", "linux"; libc="glibc")]
+cuda_platforms = [
+    # Platform("x86_64", "Linux"; cuda = "10.2"),
+    Platform("x86_64", "Linux"; cuda = "11.3"),
+]
+platforms = expand_cxxstring_abis(cuda_platforms)
+
 
 # The products that we will ensure are always built
 products = [
-    LibraryProduct("libmagma", :libmagma; dont_dlopen=true),
-    LibraryProduct("libmagma_sparse", :libmagma_sparse; dont_dlopen=true)
+    LibraryProduct("libmagma", :libmagma),
+    LibraryProduct("libmagma_sparse", :libmagma_sparse)
 ]
 
 # Dependencies that must be installed before this package can be built
 dependencies = [
-    BuildDependency(PackageSpec(name="CUDA_full_jll", version=v"11.6.0")),
-    Dependency("libblastrampoline_jll")
+    # You can only specify one cuda version in the deps. To build against more than 
+    # one cuda version, you have to include them as Archive Sources. (see Torch_jll)
+    RuntimeDependency(PackageSpec(name="CUDA_Runtime_jll")),
+    BuildDependency(PackageSpec(name="CUDA_full_jll", version=v"11.0.3"), platforms=platforms),
+    Dependency("libblastrampoline_jll"),
+    Dependency(PackageSpec(name="CompilerSupportLibraries_jll", uuid="e66e0078-7015-5450-92f7-15fbd957f2ae"))
 ]
 
 # Build the tarballs, and possibly a `build.jl` as well.
-build_tarballs(ARGS, name, version, sources, script, platforms, products, dependencies; julia_compat="1.8")
+build_tarballs(ARGS, name, version, sources, script, platforms, products, dependencies; 
+                preferred_gcc_version=v"8", 
+                julia_compat="1.6",
+                augment_platform_block=CUDA.augment)
```

Comment:
```suggestion
                augment_platform_block=CUDA.augment)
```
I would recommend to always end any source files with a newline